# IS 362 - Project 2 - Part I
## Salary Survey

The following analysis was conducted using the salary survey results containing more than 32,000 responses.


In [66]:
import pandas as pd
import numpy as np

salarysurvey = pd.read_csv('salary_survey2019.csv', engine='python')
#Salary of different currency cannot be compared so we will focus on US salary by filtering only USD currency
USsalary = salarysurvey[(salarysurvey["Currency"]=="USD")]
USsalary.head(5)

Timestamp       Age                      Industry      Jobtitle  \
0  4/26/19 17:58  under 18                      Coaching         Coach   
1   4/27/19 0:25  under 18          Contract Engineering        Owner    
3  5/11/19 11:49  under 18                     eCommerce        Seller   
4   4/29/19 1:02  under 18                     education       student   
6  4/26/19 12:50  under 18  Education Auxiliary Programs  Site Manager   

     Salary Currency                     Location      Experience  \
0   55999.0      USD                     Illinois       5-7 years   
1   18000.0      USD                    Colorado   1 year or less   
3  100000.0      USD                          NaN     2 - 4 years   
4   12000.0      USD                          NaN  1 year or less   
6    7500.0      USD  Morristown, New Jersey, USA  1 year or less   

                Notes Unnamed: 9  
0                 NaN        NaN  
1      Self employed         NaN  
3                 NaN        NaN  
4  Highschool student        NaN  
6                 NaN        NaN

In [70]:
USsalary['Salary'].mean()

103872.22388365329

The average salary of the survey respondants was $103,872. 

This result is highly unrealistic because according to the article below from The Street, the average US income in 2019 was $48, 672.
This means that either most of the survey takers earn above average income or they were not truthful on their responses.

https://www.thestreet.com/personal-finance/average-income-in-us-14852178

## Top 10 highest salary in the US according to this survey

In [42]:
HighestSalary = USsalary.sort_values('Salary', ascending=False)
df = pd.DataFrame(HighestSalary)
output = (df[['Industry', 'Jobtitle', 'Salary']])
output.head(10)

Industry                   Jobtitle  \
1566                       Architecture      Senior Vice President   
17973                  Law Enforcement               Special Agent   
7169   Education / K-12 School District                 HR Manager   
3041                      Biotechnology  Senior Research Associate   
17738                               Law            Staff Law Clerk   
16780                                IT            Systems Analyst   
16111                        Insurance   Finance Analytics Manager   
16110                        Insurance   Finance Analytics Manager   
13443                         Higher Ed        Associate Professor   
7935     Engineering/ science/ research        electrical engineer   

            Salary  
1566   $990,000.00  
17973   $99,960.00  
7169    $99,960.00  
3041    $99,950.00  
17738   $99,927.00  
16780   $99,912.98  
16111   $99,885.00  
16110   $99,885.00  
13443   $99,873.00  
7935    $99,860.00

## Top 10 lowest salary in the US according to this survey

In [71]:
LowestSalary = USsalary.sort_values('Salary', ascending=True)
df = pd.DataFrame(LowestSalary)
output = (df[['Industry', 'Jobtitle', 'Salary']])
output.head(10)

Industry  \
31709                                           Hospital   
14                                 Images and Data Entry   
32130                               on-campus employment   
14457                                      Retail Gocery   
31269  Currently a student, but I work in my drama de...   
32404                                           Software   
31259                                        County govt   
973                                              Grocery   
16453                                           Academia   
972                                     graphic designer   

                              Jobtitle  Salary  
31709  Medical Laboratory Technologist  1000.0  
14       Seasonal data entry associate  2000.0  
32130          Borrow Direct Assistant  2000.0  
14457                        Superviso  2240.0  
31269    Theatrical Journeyman/student  2900.0  
32404               Software developer  3000.0  
31259                           Intern  3000.0  
973                  Front end cashier  3150.0  
16453                          Adjunct  3500.0  
972                   graphic designer  3500.0

The ten lowest salaries are way to small, leading to the conclusion that some of the survey takers are part-time workers or they confused the annual salary with monthly or weekly pay.

## Salary analysis by age group

In [76]:
data = USsalary[(USsalary["Age"]=="55-64")]
age_set = data.sort_values('Salary', ascending=False)
df = pd.DataFrame(age_set)
output = (df[['Age', 'Industry', 'Jobtitle', 'Salary']])
output.head()

Age        Industry                     Jobtitle     Salary
1030  55-64      Healthcare                          CEO  2645000.0
805   55-64  Entertainment   Sr Director, Human Reources  1470000.0
1804  55-64        Software              Support manager  1400000.0
1369  55-64           Legal                      Partner  1100000.0
599   55-64      Consulting                      Partner  1000000.0

In [77]:
data1 = USsalary[(USsalary["Age"]=="45-54")]
age_set1 = data1.sort_values('Salary', ascending=False)
df = pd.DataFrame(age_set1)
output = (df[['Age', 'Industry', 'Jobtitle', 'Salary']])
output.head()

Age                      Industry                          Jobtitle  \
4874  45-54  Medical Device Manufacturing  End User Computing Administrator   
4443  45-54                           Law                           Partner   
4444  45-54                           Law                           Partner   
5617  45-54                 Self employed                           Founder   
3275  45-54           Financial Services                            Partner   

          Salary  
4874  82000000.0  
4443   5800000.0  
4444   3750000.0  
5617   3334000.0  
3275   2120000.0

In [78]:
data2 = USsalary[(USsalary["Age"]=="35-44")]
age_set2 = data2.sort_values('Salary', ascending=False)
df = pd.DataFrame(age_set2)
output = (df[['Age', 'Industry', 'Jobtitle', 'Salary']])
output.head()

Age        Industry                 Jobtitle     Salary
7182   35-44  Bar/restaurant                    Owner  2000000.0
9430   35-44      Government          Health Director  1750000.0
15356  35-44      Technology  Director of Engineering  1500000.0
8907   35-44         Finance                   Trader  1500000.0
9988   35-44      Healthcare      Physician assistant  1485000.0

In [79]:
data3 = USsalary[(USsalary["Age"]=="25-34")]
age_set3 = data3.sort_values('Salary', ascending=False)
df = pd.DataFrame(age_set3)
output = (df[['Age', 'Industry', 'Jobtitle', 'Salary']])
output.head()

Age                Industry                 Jobtitle     Salary
22988  25-34  Information Technology  Chief Financial Officer  2500000.0
28513  25-34                Software                  Partner  2500000.0
23420  25-34                      IT       Software Engineer   1580000.0
30501  25-34                     NaN                    I ggg  1000000.0
19590  25-34                  Energy   Director of marketing    850000.0